### PR TITLE
fix env vars since they collide with kubernetes

### PR DESCRIPTION
**Pull request by antonioberben**
**Please provide a description of this PR:**
The previous PR introduced an issue.
The env vars (i.e. DETAILS_PORT) is used by kubernetes. It was needed to use different env var to make it work in kubernetes with default port.

Below the env vars when deploying productpage on kubernetes. Notice that `DETAILS_PORT` is populated already by kubernetes. This is due to [this](https://kubernetes.io/docs/concepts/services-networking/service/#environment-variables). Easiest solution is to change the name for the envvar.

```
I have no name!@productpage-v1-5fcf6f85b-88c8f:/opt/microservices$ env | grep -i details_port
DETAILS_PORT_9080_TCP_PORT=9080
DETAILS_PORT=tcp://10.1.76.19:9080
DETAILS_PORT_9080_TCP_PROTO=tcp
DETAILS_PORT_9080_TCP=tcp://10.1.76.19:9080
DETAILS_PORT_9080_TCP_ADDR=10.1.76.19
```

### CODE REVIEW

File: samples/bookinfo/src/productpage/productpage.py [hunk 1]
```diff
@@ -59,11 +59,11 @@
 
 servicesDomain = "" if (os.environ.get("SERVICES_DOMAIN") is None) else "." + os.environ.get("SERVICES_DOMAIN")
 detailsHostname = "details" if (os.environ.get("DETAILS_HOSTNAME") is None) else os.environ.get("DETAILS_HOSTNAME")
-detailsPort = "9080" if (os.environ.get("DETAILS_PORT") is None) else os.environ.get("DETAILS_PORT")
+detailsPort = "9080" if (os.environ.get("DETAILS_APP_PORT") is None) else os.environ.get("DETAILS_APP_PORT")
```
Comment (frankbu) [inline samples/bookinfo/src/productpage/productpage.py:62]: Why aren't you just using `DETAILS_SERVICE_PORT` which is the actual port of the details `Service`?
```suggestion
detailsPort = "9080" if (os.environ.get("DETAILS_SERVICE_PORT") is None) else os.environ.get("DETAILS_SERVICE_PORT")
```

Comment (antonioberben) [reply samples/bookinfo/src/productpage/productpage.py:62]: makes sense

**Review by GregHanson — COMMENTED** [review]
this will require new bookinfo images to get built correct?

**Comment by antonioberben** [comment]
@GregHanson , I would say yes. But not sure how that is automated :(

**Comment by ericvn** [comment]
> this will require new booking images to get built correct?

Only if you want this behavior when using samples from a release. One can always build/use their own images.

@frankbu is the person who has to handle any sample image builds.

**Comment by antonioberben** [comment]
Hi @GregHanson, are you going to create a new image or is it enough to just approve it? Thanks

**Comment by antonioberben** [comment]
@GregHanson , understood. Thanks! If it's needed first to be merged, please, could you (or @frankbu) approve this PR? Thanks

**Comment by ericvn** [comment]
@antonioberben Have you done the testing yourself as noted in the [readme](https://github.com/istio/istio/blob/master/samples/bookinfo/README.md):  `Note: We need the owner of the PR to perform the appropriate testing with built/pushed images to their own docker repository before we would build/push images to the official Istio repository.`

**Comment by antonioberben** [comment]
Yes, we use images based on this code in a workshop recently

**Comment by ericvn** [comment]
> Yes, we use images based on this code in a workshop recently

Then we need to wait for @frankbu to build new images. I do know there are a few other PRs out there (automated so not getting a pre-test run), and not knowing if there is a reason to wait for someone to verify those and pull into one big image update.

**Review by ericvn — APPROVED** [review]
I am approving a set of PRs to include in a new set of samples images.

**Pull request merged**